### PR TITLE
Update open-source-services.md - opensourcing tests from legal perspective

### PR DESCRIPTION
**Pull request by ohudlick**
Adding a short summary, what type of tests cannot be shared to community due to legal issues.
Based on guidance gathered previously from Red Hat legal team.

### CODE REVIEW

File: open-source-services.md
Comment (stefwalter) [inline open-source-services.md:43]: Can we replace "internal" with "company private". 

Why do we list "tests for CVE" separate here? Doesn't embargoed cover the topic? Don't we want tests that verify a CVE fix to be available publically once that CVE is public and no longer embargoed, right?

Why do testing performance, failover, data recovery and compliance with standards tests violate _legal restrictions_?

**Review by stefwalter — CHANGES_REQUESTED** [review]

File: open-source-services.md
Comment (schwesig) [reply open-source-services.md:43]: +1 company private sounds good.
Do we at Red Hat have something like these tier levels as Top Secret, Secret, Confidential, as I see in some presentations? Maybe we could use that level to define what can be shared and what not?

File: open-source-services.md
Comment (schwesig) [inline open-source-services.md:45]: +1

File: open-source-services.md
Comment (quaid) [reply open-source-services.md:43]: This is a fix just for what seems clear here so far:

```suggestion
Tests that violate _legal restrictions_ cannot be shared publicly.
Specifically: tests that include references to company and customer private information, URLs, IP addresses, configuration or other data; tests for CVE, embargoed or private bugs; testing performance, failover, data recovery and compliance with security standards (like FIPS). 
```

Comment (quaid) [reply open-source-services.md:43]: > Can we replace "internal" with "company private".

Addressed in my suggestion.

> Why do we list "tests for CVE" separate here? Doesn't embargoed cover the topic? Don't we want tests that verify a CVE fix to be available publically once that CVE is public and no longer embargoed, right?

I agree there is nuance here that includes some level and time of openness; perhaps just "ermbargoed or private bugs" covers all the cases, and we can work out the specific details within a security SIG.

> Why do testing performance, failover, data recovery and compliance with standards tests violate _legal restrictions_?

Good question, I can make some guesses, but we need some fairly concrete explanations of why something is kept private/secret.

File: open-source-services.md
Comment (quaid) [reply open-source-services.md:45]:
```suggestion
It is _recommended_ other tests that are independent of the service (e.g. penetration tests) are Open Source, but this is not required.
```

**Review by quaid — CHANGES_REQUESTED** [review]
I made some suggested fixes, which @ohudlick can commit directly from here, and added some thoughts to @stefwalter's quesitons.

File: open-source-services.md
Comment (ohudlick) [reply open-source-services.md:43]: I am reaching to expert in legal team to provide answer, stay tuned.

**Comment by quaid** [comment]
Holding for @ohudlick to get back and resolve the open discussion.
/hold

File: open-source-services.md
Comment (ondrahu) [reply open-source-services.md:43]: I got response from RH legal team I'll reply comments and incorporate the changes by April 20.

Comment (ohudlick) [reply open-source-services.md]: Comments and commits incorporated, please review again

**Comment by stefwalter** [comment]
@ohudlick Do you have time to look at the review? Once you do, we can get this merged (and reopen it). WDYT?

**Pull request closed without merging**
